src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+ */
+export type TrackDisplayMode =
+  | 'none'
+  | 'odd-segments'
+  | 'even-segments'
+  | 'undefined';
+
+/** The interaction mode. Default is 1. Possible values:
+ * 'allow-crossing' - allows thumb to cross other thumbs. Default.
+ * 'step-separated'- keeps the thumb from crossing and separated by a step.
+ */
+export type ThumbMoveMode = 'allow-crossing' | 'step-separated';
+
+/** Local Context set up by Slider and provided to children */
+interface SliderContextProps {
+  readonly values: number[];
+  readonly step: number;
+  readonly trackDisplayMode: TrackDisplayMode;
+  readonly railDomain: ValueDomain;
+  readonly thumbMode: ThumbMoveMode;
+  readonly toolTipPosition: Placement;
+  readonly setThumbMoveIndex: (index: number) => void;
+  readonly setThumbValue: (index: number, value: number) => void;
+  readonly toolTipFunction?: (val: number, step: number) => string;
+  readonly hideTooltip: boolean;
+}
+
+const SliderContext = React.createContext<SliderContextProps>({
+  values: [],
+  step: 1,
+  trackDisplayMode: 'none',
+  hideTooltip: false,
+  railDomain: { min: 0, max: 100 },
+  thumbMode: 'allow-crossing',
+  toolTipPosition: 'top',
+  setThumbMoveIndex: () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  },
+  setThumbValue: () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  },
+});
+
+const useSliderContext = (): SliderContextProps => {
+  return React.useContext(SliderContext);
+};
+
+/** Private Utility Functions */
+function clamp(val: number, min: number, max: number) {
+  return val > max ? max : val < min ? min : val;
+}
+
+function getPercentageOfRectangle(rect: DOMRect, pointer: number) {
+  const position = clamp(pointer, rect.left, rect.right);
+  return (position - rect.left) / rect.width;
+}
+
+function getAllowableThumbRange(
+  thumbMode: ThumbMoveMode,
+  values: number[],
+  index: number,
+  railDomain: ValueDomain,
+  step: number,
+) {
+  if (thumbMode === 'step-separated') {
+    const min = index === 0 ? railDomain.min : values[index - 1] + step;
+    const max =
+      index < values.length - 1 ? values[index + 1] - step : railDomain.max;
+    return [min, max];
+  }
+  return [railDomain.min, railDomain.max];
+}
+
+function getClosestValueIndex(values: number[], pointerValue: number) {
+  if (1 === values.length) {
+    return 0;
+  }
+  const distances = values.map((value) => Math.abs(value - pointerValue));
+  const smallest = Math.min(...distances);
+  return distances.indexOf(smallest);
+}
+
+function getDefaultTrackDisplay(
+  trackDisplayMode?: TrackDisplayMode,
+  values?: number[],
+): TrackDisplayMode {
+  const numValues = values?.length ?? 0;
+  if (undefined !== trackDisplayMode) {
+    return trackDisplayMode;
+  }
+
+  return numValues % 2 ? 'odd-segments' : 'even-segments';
+}
+
+function roundValueToClosestStep(value: number, step: number) {
+  const positiveStep = Math.abs(step);
```

Comment:
How step can be negative? Isnt it just a positive number??

Reply:
I was thinking of another use case, for the slider it should always be positive. I will update.